contexts/ActionsContext.tsx

```diff
@@ -0,0 +1,97 @@
+import { useDisclosure } from '@chakra-ui/react';
+import { createContext, useContext, useMemo, useState } from 'react';
+
+import { useGame } from '@/contexts/GameContext';
+import { Character } from '@/utils/types';
+
+enum PlayerActions {
+  EDIT = 'Edit',
```

Comment:
I like the idea. But don't you think this context will hold too many functions once all the actions are moved here ? 

Merging for now anyways :)